teams_rules.go

```diff
@@ -0,0 +1,222 @@
+	OverrideHost string `json:"override_host"`
+
+	// settings for l4(network) level overrides
+	L4Override *TeamsL4OverrideSettings `json:"l4override"`
+
+	// settings for browser isolation actions
+	BISOAdminControls *TeamsBISOAdminControlSettings `json:"biso_admin_controls"`
+}
+
+// TeamsL4OverrideSettings used in l4 filter type rule with action set to override
+type TeamsL4OverrideSettings struct {
+	IP   string `json:"ip,omitempty"`
+	Port int    `json:"port,omitempty"`
+}
+
+type TeamsBISOAdminControlSettings struct {
+	DisablePrinting  bool `json:"dp"`
+	DisableCopyPaste bool `json:"dcp"`
+}
+
+type TeamsFilterType string
+
+type TeamsGatewayAction string
+
+const (
+	HttpFilter TeamsFilterType = "http"
+	DnsFilter  TeamsFilterType = "dns"
+	L4Filter   TeamsFilterType = "l4"
+)
+
+const (
+	Allow        TeamsGatewayAction = "allow"
+	Block        TeamsGatewayAction = "block"
+	SafeSearch   TeamsGatewayAction = "safesearch"
+	YTRestricted TeamsGatewayAction = "ytrestricted"
+	On           TeamsGatewayAction = "on"
+	Off          TeamsGatewayAction = "off"
+	Scan         TeamsGatewayAction = "scan"
+	NoScan       TeamsGatewayAction = "noscan"
+	Isolate      TeamsGatewayAction = "isolate"
+	NoIsolate    TeamsGatewayAction = "noisolate"
+	Override     TeamsGatewayAction = "override"
+	L4Override   TeamsGatewayAction = "l4_override"
+)
+
+// TeamsRule represents an Teams wirefilter rule.
+type TeamsRule struct {
+	ID          string             `json:"id,omitempty"`
+	CreatedAt   *time.Time         `json:"created_at,omitempty"`
+	UpdatedAt   *time.Time         `json:"updated_at,omitempty"`
+	DeletedAt   *time.Time         `json:"deleted_at,omitempty"`
+	Name        string             `json:"name"`
+	Description string             `json:"description"`
+	Precedence  uint64             `json:"precedence"`
+	Enabled     bool               `json:"enabled"`
+	Action      TeamsGatewayAction `json:"action"`
+	Filters     []TeamsFilterType  `json:"filters"`
+	Traffic     string             `json:"traffic"`
+	Identity    string             `json:"identity"`
+	Version     uint64             `json:"version"`
+	Settings    TeamsRuleSettings  `json:"rule_settings,omitempty"`
+}
+
+// TeamsRuleResponse is the API response, containing a single rule
+type TeamsRuleResponse struct {
+	Response
+	Result TeamsRule `json:"result"`
+}
+
+// TeamsRuleResponse is the API response, containing an array of rules
+type TeamsRulesResponse struct {
+	Response
+	Result []TeamsRule `json:"result"`
+}
+
+// TeamsRulePatchRequest is used to patch an existing rule
+type TeamsRulePatchRequest struct {
+	UUID        string             `json:"id"`
+	Name        string             `json:"name"`
+	Description string             `json:"description"`
+	Precedence  uint64             `json:"precedence"`
+	Enabled     bool               `json:"enabled"`
+	Action      TeamsGatewayAction `json:"action"`
+	Settings    TeamsRuleSettings  `json:"rule_settings,omitempty"`
```

Comment:
```suggestion
	ID        string             `json:"id"`
	Name        string             `json:"name"`
	Description string             `json:"description"`
	Precedence  uint64             `json:"precedence"`
	Enabled     bool               `json:"enabled"`
	Action      TeamsGatewayAction `json:"action"`
	RuleSettings    TeamsRuleSettings  `json:"rule_settings,omitempty"`
```

these should also match the JSON tag names